src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UsageReport/WriteUsageBurndownData.cs

```diff
@@ -43,30 +44,41 @@ public class WriteUsageBurndownData : Task
         [Required]
         public string OutputFilePath { get; set; }
 
-        public override bool Execute()
+        /// <summary>
+        ///  Sends HTTP requests and receives HTTP responses.
+        /// </summary>
+        private readonly HttpClient client = new HttpClient();
+
+        public override bool Execute() => ExecuteAsync().GetAwaiter().GetResult();
+
+        private async Task<bool> ExecuteAsync()
         {
             string baselineRelativeFileName = PrebuiltBaselineFile.Replace(RootDirectory, "");
             string gitLogCommand = $"log --first-parent --pretty=format:%H,%f,%ci -- {PrebuiltBaselineFile}";
 
             DateTime startTime = DateTime.Now;
             Log.LogMessage(MessageImportance.High, "Generating summary usage burndown data...");
 
-            ParallelQuery<string> data = ExecuteGitCommand(RootDirectory, gitLogCommand).AsParallel().Select(commitLine =>
-            {
-                var splitLine = commitLine.Split(',');
-                var commit = new Commit()
+
+            IEnumerable<Task<Commit>> getCommitTasks = ExecuteGitCommand(RootDirectory, gitLogCommand)
+                .Select(async commitLine =>
                 {
-                    Sha = splitLine[0],
-                    Title = splitLine[1],
-                    CommitDate = DateTime.Parse(splitLine[2])
-                };
-                string fileContents = GetFileContents(baselineRelativeFileName, commit.Sha);
-                Usage[] usages = UsageData.Parse(XElement.Parse(fileContents)).Usages.NullAsEmpty().ToArray();
-                commit.PackageVersionCount = usages.Count();
-                commit.PackageCount = usages.GroupBy(i => i.PackageIdentity.Id).Select(grp => grp.First()).Count();
-                return commit;
-            })
-            .Select(c => c.ToString());
+                    var splitLine = commitLine.Split(',');
+                    var commit = new Commit()
+                    {
+                        Sha = splitLine[0],
+                        Title = splitLine[1],
+                        CommitDate = DateTime.Parse(splitLine[2])
+                    };
+                    string fileContents = await GetFileContentsAsync(baselineRelativeFileName, commit.Sha);
+                    Usage[] usages = UsageData.Parse(XElement.Parse(fileContents)).Usages.NullAsEmpty().ToArray();
+                    commit.PackageVersionCount = usages.Count();
+                    commit.PackageCount = usages.GroupBy(i => i.PackageIdentity.Id).Select(grp => grp.First()).Count();
+                    return commit;
+                });
+
+            Commit[] commits = await System.Threading.Tasks.Task.WhenAll(getCommitTasks);
```

Comment:
Is the fully qualified type necessary here?

```suggestion
            Commit[] commits = await Task.WhenAll(getCommitTasks);
```